hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/helpers/TestContainerUtils.java

```diff
@@ -122,6 +127,23 @@ public void testDatanodeIDPersistent(@TempDir File tempDir) throws Exception {
     assertWriteRead(tempDir, id1);
   }
 
+  @Test
+  public void testVerifyInconsistentChecksumShouldFail() throws Exception {
+    ContainerData containerData = mock(ContainerData.class);
+    when(containerData.getContainerType()).thenReturn(
+        ContainerProtos.ContainerType.KeyValueContainer);
+    // mock to return different checksum
+    when(containerData.getChecksum()).thenReturn("checksum1", "checksum2");
+    conf.setBoolean(HddsConfigKeys.HDDS_CONTAINER_CHECKSUM_VERIFICATION_ENABLED, true);
+    try {
+      ContainerUtils.verifyChecksum(containerData, conf);
+      fail("code should not reach here, exception should have been thrown.");
+    } catch (Exception e) {
+      assertTrue(e.getMessage().contains("Container checksum error"));
+    }
+  }
```

Comment:
We already have this test case:

https://github.com/apache/ozone/blob/422df39669f424d62e98de061be6b90190907e91/hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/impl/TestContainerDataYaml.java#L289-L301

New functionality is in `ContainerImporter#importContainer`, should be tested in `TestContainerImporter`.

Reply:
yeah agree! updated in TestContainerImporter.